### PR TITLE
Adding ProjectClass.cs to DistroRickDev/PMS-Core Repository

### PR DESCRIPTION
**Pull request by ivanmartin91**
ProjectClass is based on the group project instructions and class diagram, it shows the connection between the project and tasks considering each project has a list of tasks.

### CODE REVIEW

File: PMSCore/ProjectClass.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            Name = name; //setting the name for it
+            Description = description; //providing a description
+        }
+
+        public void UpdateStatus(string newStatus) //method that updates a new value for this task => abstraction (status can't be changed directly)
+        {
+            if (newStatus == "ToDo" || newStatus == "In Progress" || newStatus == "Completed") //update status only if it matches one of these: "ToDo", "In Progress", "Completed"
+                Status = newStatus;
+        }
+    }
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: This is not part of this ticket nor this PR there is work in progress for the TaskClass, if you need it you should create an empty one

File: PMSCore/ProjectClass.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    //class Project which is a group of tasks, also its attributes and methods are another example of Abstraction
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: Please let's keep the class comment documentation unifrom please see this example:
https://github.com/DistroRickDev/PMS-Core/pull/7/files
Also no need to inline comment every code line the code speaks for itself

File: PMSCore/ProjectClass.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    public class Project { 
+        public string Name { get; set; } //attribute to give this project a name
+        public string Description { get; set; } //providing the project's description
+        public DateTime CreationDate { get; } = DateTime.Now; //creation date, according to the instructions that each project will have a timestamp of its creation
+        public DateTime Deadline { get; set; } //instructions state that the project needs to have a deadline in addition to a name and description
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: What about project Status? 
Also suggest creating an Enum called ProjectStatus for it

File: PMSCore/ProjectClass.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            if (newStatus == "ToDo" || newStatus == "In Progress" || newStatus == "Completed") //update status only if it matches one of these: "ToDo", "In Progress", "Completed"
+                Status = newStatus;
+        }
+    }
+    //class Project which is a group of tasks, also its attributes and methods are another example of Abstraction
+    public class Project { 
+        public string Name { get; set; } //attribute to give this project a name
+        public string Description { get; set; } //providing the project's description
+        public DateTime CreationDate { get; } = DateTime.Now; //creation date, according to the instructions that each project will have a timestamp of its creation
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: this should be initialized like this in the costructor I would say

**Review by DistroRickDev — COMMENTED** [review]
Looks solid but other then the comments left I have a couple aspects I would like you to consider:

- Class you created should be unit tested in this PR
- Project should have a status.
- Maybe an internal string that appends every action as a log (so in the end a report of the project can be generated) what do you think

**Comment by ivanmartin91** [comment]
@DistroRickDev , just updated the ProjectClass.cs based on your feedback, please have a look when you get a chance. Thanks!

**Comment by DistroRickDev** [comment]
> @DistroRickDev , just updated the ProjectClass.cs based on your feedback, please have a look when you get a chance. Thanks!

@ivanmartin91  The project class was ok you need to add a new file in the test Directory to test for the class created as done here: https://github.com/DistroRickDev/PMS-Core/pull/8

**Comment by ivanmartin91** [comment]
@DistroRickDev , adding a new file in PMSCore.Test for the class unit tests and also reverting the code to the version without unit tests in PMSCore, hope this is okay now.

File: PMSCore/ProjectClass.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Generates a report of all actions performed on the project.
+        /// </summary>
+        public void GenerateReport()
+        {
+            Console.WriteLine("Action Log:");
+            Console.WriteLine(actionLog);
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: maybe have something like this instead:
```c#
private string report;
private void addToReport(string message)
{
   report += $"[ActionLog] ${DateTime.Now().ToString()} : ${message}"
}
public GenerateReport() 
{
   return _report;
}
```

File: PMSCore/ProjectClass.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Represents a task to be defined later.
+    /// </summary>
+    public class Task { }
```
Comment (DistroRickDev) [inline PMSCore/ProjectClass.cs]: SOLID principles, singe responsability purpose each class should have one responsability, please movr this to Task.cs (every file is treated as class)

**Review by DistroRickDev — COMMENTED** [review]
Great work only a couple of final suggestions, and please squash the commits

**Comment by ivanmartin91** [comment]
Hey @DistroRickDev , just removed those 4 lines and put them as a separate Task.cs, also added addToReport private string report Field and methods addToReport() and GenerateReport(). Finally tried to squash those two commits but it didn't work: tried by installing GitHub desktop, using Terminal and also from the Git Repository section by right clicking but ''squash commits'' is greyed out (see attached).

**Comment by DistroRickDev** [comment]
Looks good to me no apparent issues found

**Review by DistroRickDev — APPROVED** [review]
LGTM

**Pull request merged**